tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,966 @@
+                if started_arguments is False:
+                    started_arguments = True
+                    bracket_count = 1
+                else:
+                    bracket_count += 1
+            elif input_string[pos] == ")" and started_arguments:
+                bracket_count -= 1
+
+            if bracket_count == 0 and started_arguments:
+                # Finished!
+                break
+            pos += 1
+
+        function_call = input_string[start:pos + 1]
+        output_string = output_string.replace(function_call, replacement)
+
+    return output_string
+
+
+def disable_module(input_file):
+    """Disable a module entirely except for header includes."""
+    with openf(input_file, "r+") as f:
+        txt = f.read()
+        last = list(re.finditer(r"#include .*\n", txt))[-1]
+        end = last.end()
+
+        disabled = "%s#if !defined(__HIP_PLATFORM_HCC__)\n%s\n#endif" % (txt[0:end], txt[end:])
+
+        f.seek(0)
+        f.write(disabled)
+        f.truncate()
+
+
+def extract_arguments(start, string):
+    """ Return the list of arguments in the upcoming function parameter closure"""
+    arguments = []
+    closures = {
+        "<": 0,
+        "(": 0
+    }
+    current_position = start
+    argument_start_pos = current_position + 1
+
+    # Search for final parenthesis
+    while current_position < len(string):
+        if string[current_position] == "(":
+            closures["("] += 1
+        elif string[current_position] == ")":
+            closures["("] -= 1
+        elif string[current_position] == "<":
+            closures["<"] += 1
+        elif string[current_position] == ">" and string[current_position - 1] != "-":
+            closures["<"] -= 1
+
+        # Finished all arguments
+        if closures["("] == 0 and closures["<"] == 0:
+            # Add final argument
+            arguments.append({"start": argument_start_pos, "end": current_position})
+            break
+
+        # Finished current argument
+        if closures["("] == 1 and closures["<"] == 0 and string[current_position] == ",":
+            arguments.append({"start": argument_start_pos, "end": current_position})
+            argument_start_pos = current_position + 1
+
+        current_position += 1
+
+    return arguments
+
+
+def add_static_casts(directory, extensions, KernelTemplateParams):
+    """Added necessary static casts to kernel launches due to issue in HIP"""
+    # Add static_casts<> to all kernel launches.
+    for (dirpath, _dirnames, filenames) in os.walk(directory):
+        for filename in filenames:
+            if filename_ends_with_extension(filename, extensions):
+                filepath = os.sep.join([dirpath, filename])
+                with openf(filepath, "r+") as fileobj:
+                    input_source = fileobj.read()
+                    new_output_source = input_source
+                    get_kernel_definitions = [k for k in re.finditer("hipLaunchKernelGGL\(", input_source)]
```

Comment:
The generator is used only once, you don't have to listify it. To make it less error prone, move the generator inside the for-loop (don't bind a name to it)

Reply:
Changed this here and in another place. There is still more but those have such complicated matches that I find the current code more readable.